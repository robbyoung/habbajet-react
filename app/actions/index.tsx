export * from './actionTypes';
export * from './addHabbajet';
export * from './addHabitResult';
export * from './updateBudget';
export * from './resetHabbajet';
export * from './loadState';
export * from './selectHabbajet';
export * from './updateEditorField';
export * from './validateEditor';
export * from './clearEditor';
