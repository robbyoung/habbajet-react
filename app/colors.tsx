export const white = '#ffffff';
export const lightGrey = 'F5F5F5';
export const grey = '#959595';

export const habbajetColors = [
    '#8066C2', // Purple
    '#2AAD5E', // Green
    '#E04747', // Red
    '#7F90EC', // Blue
    '#B1B952', // Gold
];

export const complementColors: {[id: string]: string} = {
    '#8066C2': '#C2A066',
    '#2AAD5E': '#AD372A',
    '#E04747': '#47E047',
    '#7F90EC': '#ECB27F',
    '#B1B952': '#B952B0',
};
