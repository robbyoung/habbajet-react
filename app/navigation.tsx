import {Navigation} from 'react-native-navigation';
import React from 'react';
import HabbajetScreen from './containers/habbajetScreen';
import LoadingScreen from './containers/loadingScreen';
import {Provider} from 'react-redux';
import store from './store';

enum Screens {
    Loading = 'Loading',
    Habbajet = 'Habbajet',
}

export const STACK_NAVIGATOR = 'StackNavigator';

Navigation.registerComponent(
    Screens.Loading,
    () => () => (
        <Provider store={store}>
            <LoadingScreen />
        </Provider>
    ),
    () => LoadingScreen,
);

Navigation.registerComponent(
    Screens.Habbajet,
    () => () => (
        <Provider store={store}>
            <HabbajetScreen />
        </Provider>
    ),
    () => HabbajetScreen,
);

export const goToLoading = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Loading,
                            options: {
                                topBar: {
                                    visible: false,
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};

export const goToHabbajet = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Habbajet,
                            options: {
                                topBar: {
                                    title: {
                                        text: ' ',
                                        fontFamily: 'Abel',
                                        fontSize: 30,
                                        color: '#ffffff',
                                    },
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};
