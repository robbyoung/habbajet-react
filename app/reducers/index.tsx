import habbajets from './habbajets';
import purchases from './purchases';
import budget from './budget';
import habbajetEditor from './habbajetEditor';
import purchaseEditor from './purchaseEditor';
import tagEditor from './tagEditor';
import tags from './tags';
import {combineReducers} from 'redux';

export default combineReducers({
    habbajets,
    purchases,
    budget,
    tags,
    habbajetEditor,
    purchaseEditor,
    tagEditor,
});
