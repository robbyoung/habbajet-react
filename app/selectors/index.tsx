export * from './budget';
export * from './purchases';
export * from './habbajets';
export * from './habbajetEditor';
