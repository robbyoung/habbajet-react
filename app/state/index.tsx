export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
