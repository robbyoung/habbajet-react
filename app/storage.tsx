import AsyncStorage from '@react-native-community/async-storage';
import {State} from './state';
import store from './store';
import {createTestState} from './state/testState';

// const defaultState: State = {
//     budget: 0,
//     habbajets: [],
//     purchases: [],
// };

const defaultState = createTestState(1, 0, 0);

export async function saveState(): Promise<void> {
    const state = store.getState();
    await AsyncStorage.setItem('state', JSON.stringify(state));
}

export async function loadState(): Promise<State> {
    const state = await AsyncStorage.getItem('state');
    let savedState: State;
    if (state !== undefined && state !== null) {
        savedState = JSON.parse(state) as State;
        return savedState;
    }
    return defaultState;
}

export async function clearState() {
    await AsyncStorage.clear();
}
