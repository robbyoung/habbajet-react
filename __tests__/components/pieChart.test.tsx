import React from 'react';
import renderer from 'react-test-renderer';
import {habbajetColors} from '../../app/colors';
import PieChart from '../../app/components/pieChart';

describe('PieChart Component', () => {
    it('renders a pie chart for the given stats', () => {
        const stats = [
            {
                tagName: 'Tag 1',
                total: 6,
                percentage: 0.6,
                color: habbajetColors[0],
            },
            {
                tagName: 'Tag 2',
                total: 3,
                percentage: 0.3,
                color: habbajetColors[1],
            },
            {
                tagName: 'Tag 3',
                total: 1,
                percentage: 0.1,
                color: habbajetColors[2],
            },
        ];

        const component = renderer.create(<PieChart sections={stats} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
