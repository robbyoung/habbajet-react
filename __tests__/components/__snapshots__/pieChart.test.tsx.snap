// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart Component renders a pie chart for the given stats 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 600,
          "width": 600,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          6.123233995736766e-17,
          -1,
          1,
          6.123233995736766e-17,
          0,
          600,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            450,
            300,
            4,
            300,
            300,
            150,
            0,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#ffffff"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={300}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            450,
            300,
            4,
            300,
            300,
            150,
            0,
            3.7699111843077517,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#8066c2"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={300}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            178.6474508437579,
            211.83221215612906,
            4,
            300,
            300,
            150,
            3.7699111843077517,
            5.654866776461628,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e04747"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={300}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            421.3525491562421,
            211.832212156129,
            4,
            300,
            300,
            150,
            5.654866776461628,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#2aad5e"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={300}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTGroup
        opacity={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;
